attributes filter

Mel-dev-hub opened the pull request:
Still need to solve the matter of disabling the opposite attribute but this is a first version

visitharri commented on frontend/src/components/MapFilters/AttributesFilter/AttributesFilter.tsx:
```diff
@@ -0,0 +1,165 @@
+        </MapFilterItem>
+    );
+};
+export default AttributesFilter;
```
I checked your translations for the names of the opposite attributes, and I think they are fine.

visitharri commented on frontend/src/components/MapFiltersMenu/MapFiltersMenu.tsx:
```diff
@@ -91,7 +99,8 @@ const MapFiltersMenu = ({
                     <NameContainsFilter onChange={modifyNameContains} nameContains={mapFilters.nameContains} eventKey="4"/>
-                    <DifficultyFilter onChange={modifyDifficulty} difficulty={mapFilters.difficulty} eventKey="6"/> 
+                    <DifficultyFilter onChange={modifyDifficulty} difficulty={mapFilters.difficulty} eventKey="6"/>
+                    <AttributesFilter onChange={modifyAttributes} attributes={mapFilters.attributes} eventKey="7"/>
```
We need to configure eventKeys and sequence. I was following Figma. So when you merge from main, we should put them in order, and use eventKeys accordingly. 1-4 are like this in Main, 5 is Description contains, 6 Difficulty, 7 Terrain. So attributes should be 8th, and I'll put Published/Hidden as 9, the last one here.

Mel-dev-hub replied:
ok

kemm0 reviewed (changes requested):
This translation should be changed:

Käytävissä ympäri vuorokauden -> Käytettävissä ympäri vuorokauden

Some icons are stretched -> add height: 20px to .icon in CacheTypeFilter.scss

Mel-dev-hub commented:
Fixed mentioned bugs. Also fixed bug when you close the side menu or press "käytä" button the disabled buttons are now correct. However, when the "tyhjennä" button is pressed there is a new bug in which the disabled buttons stay there until you either close the side menu or press the "tyhjennä" button.

visitharri commented:
There are quite a lot changes merged to main today about functionality of "käytä" and "tyhjennä" buttons. So possibly need to merge from main, and check after that how they behave.

Mel-dev-hub commented:
Everything seems to be working fine now

visitharri commented on frontend/src/components/MapFiltersMenu/MapFiltersMenu.tsx:
```diff
@@ -147,9 +154,11 @@ const MapFiltersMenu = ({
+                    <DifficultyFilter onChange={modifyDifficulty} difficulty={mapFilters.difficulty} eventKey="6"/>
                     <CacheDescriptionFilter onChange={modifyDescriptionContains} description={mapFilters.description} eventKey="5"/> 
                     <DifficultyFilter onChange={modifyDifficulty} difficulty={mapFilters.difficulty} eventKey="6"/>
                     <TerrainFilter onChange={modifyTerrain} terrain={mapFilters.terrain} eventKey="7"/> 
```
DifficultyFilter is 2 times here, You should remove line 157

visitharri replied:
Difficulty filter is 2 times here. You should remove line 157

visitharri commented:
After removing that line (2f05be7) unit test for difficulty works again. So you can remove commenting it out. I tried that unit test in my local and it's working now
